lib/rubber/cloud/datto.rb

```diff
@@ -0,0 +1,97 @@
+require 'rubber/cloud/generic'
+require 'httparty'
+
+module Rubber
+  module Cloud
+    class Datto < Base
+      def initialize(env, capistrano)
+        super(env, capistrano)
+      end
+
+      def endpoint
+        env.endpoint
+      end
+
+      def active_state
+        'running'
+      end
+
+      def stopped_state
+        'stopped'
+      end
+
+      def create_instance(instance_alias, image_name, image_type, security_groups, availability_zone, datacenter)
+        response = HttpAdapter.post(uriBuilder)
+        if response.code != 200
+          raise errFromResponse(response)
+        end
+
+        response["id"]
+      end
+
+      def describe_instances(instance_id=nil)
+        instances = []
+        if instance_id.nil?
+          response = HttpAdapter.get(uriBuilder)
+          if response.code != 200
+            raise errFromResponse(response)
+          end
+
+          response["results"].each do |item|
+            instances << instance_from_response(item)
+          end
+        else
+          response = HttpAdapter.get(uriBuilder(instance_id))
+          if response.code != 200
+            raise errFromResponse(response)
+          end
+
+          instances << instance_from_response(response)
+        end
+        instances
+      end
```

Comment:
won't work due to the differences in the response =( lets discuss though when you get in.